### DISCUSSION TITLE
Create ATAC-pseudobulk from a list of cells for peak-gene links

### DISCUSSION THREAD

**Discussion by GrafZahl1234** [GreenleafLab/ArchR#1754] [Questions / Documentation] [answered]
Hi all,

I have worked myself through the manual and now have my PeakToGeneLinks. I have now generated knn-based lists of cell labels to make pseudo-bulks.

My first question now is: Is there a simple/inbuilt way to aggregate these cell lists to pseudo-bulk-peaks? Preferably I would want to use the peak regions found using "10.1 The Iterative Overlap Peak Merging Procedure".

I know there is a categorization of peaks in Distal, Exonic, Intronic and Promoter, but I did not find a way how to get which peak belongs to which category (I would assume that is also gene specific) except a old homebrew solution (https://github.com/GreenleafLab/ArchR/discussions/838)

My second question is: Is there a inbuilt way to get this information for each peak-gene-link nowadays? It doesn't show up as a column when I use getPeak2GeneLinks. 

Lastly, there is no category “gene body”, in the peaks list or at least I found none.

My last question is how to create such pseudo-bulk-peaks (as described in question 1) for specific regions, for example the gene bodies of all genes. Can this be done in ArchR, or do I need to use MACS2 for that.

Many thanks in advance!

**Comment by rcorces** [comment, marked as answer]
> My first question now is: Is there a simple/inbuilt way to aggregate these cell lists to pseudo-bulk-peaks?

Add a new columne to cellColData defining whatever cell groupings you want and then provide that to `addReproduciblePeakSet()` `groupby` param.

> I know there is a categorization of peaks in Distal, Exonic, Intronic and Promoter, but I did not find a way how to get which peak belongs to which category (I would assume that is also gene specific) except a old homebrew solution (https://github.com/GreenleafLab/ArchR/discussions/838)

We dont provide functions for every operation one might want to perform. Some things like this just require a little manual code to perform. All of the info is there, as shown in the post you linked.

> Lastly, there is no category “gene body”, in the peaks list or at least I found none.

Gene body is not a relevant category beyond exonic and intronic.

> My last question is how to create such pseudo-bulk-peaks (as described in question 1) for specific regions, for example the gene bodies of all genes. Can this be done in ArchR, or do I need to use MACS2 for that.

No. You would create the peak set as described above and then subset that peak set to your regions of interest

**Comment by GrafZahl1234** [comment]
Thank you for your answer!
